### PR TITLE
Fixed links to examples, added examples/README.md, improved some names.

### PR DESCRIPTION
**Pull request by Martinsos**
This PR is inspired by https://github.com/wasp-lang/wasp/commit/5e45dad54a5efaf1f4dd74d9a047b72b541f3c79, kind of a follow-up on it.

- I added examples/README.md, that explains for each example what it does and why it is interesting. I know you tried to do it with the names, but I don't think that works as well as it should due to very little space.
- I renamed `gpt-response-streaming` to be just `streaming` since it doesn't do anything with gpt, it just does HTTP streaming of dummy data.
- I fixed some small mistakes/typos in the example cards on the landing page.
- I fixed links on the web/docs to examples to point to `release`, not `main`, since `release` is in sync with the latest released Wasp version, while `main` can contain whatever.
- I reverted names for `waspello`, `waspleau`, `thoughts`, because I don't think those bigger names are needed any more now that we have README.md . But, if you think it is still valuable to have more descriptive names, I would suggest following: `waspello_the-trello-clone`, `waspleau_the-dashboard-app`, `thoughts_the-note-taking-app`.

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+**tutorials** dir contains [Wasp tutorial](https://wasp-lang.dev/docs/tutorial/create) apps.
+
+1. **todo-typescript**
+    - A simple task tracking app implemented in TypeScript.
+    - Demonstrates: **full-stack type safety in Wasp via TypeScript**, [auth](https://wasp-lang.dev/docs/auth/overview), [rpc](https://wasp-lang.dev/docs/data-model/operations/overview)
+
+2. **waspello**
+    - A simple Trello clone.
+    - Demonstrates: **[optimistic updates](https://wasp-lang.dev/docs/data-model/operations/actions#the-useaction-hook-and-optimistic-updates)**, [auth](https://wasp-lang.dev/docs/auth/overview), [rpc](https://wasp-lang.dev/docs/data-model/operations/overview)
```
Comment (sodic) [inline examples/README.md:15]: Unless someone implemented this without me noticing, I don't believe it does :D

File: examples/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+Example Wasp apps
+=================
+
+These are officially maintained example Wasp apps!
```
Comment (sodic) [inline examples/README.md:4]:
```suggestion
Here's a list of all officially maintained Wasp example apps!
```
Conveys the same message without relying on the text that follows.

File: examples/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+Example Wasp apps
+=================
+
+These are officially maintained example Wasp apps!  
+Most of them are relatively small and demonstrate a couple of interesting Wasp features.
```
Comment (sodic) [inline examples/README.md:5]:
```suggestion
Most of these apps are relatively small, and each one demonstrates several of Wasp's interesting features.
```

File: examples/README.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+**tutorials** dir contains [Wasp tutorial](https://wasp-lang.dev/docs/tutorial/create) apps.
```
Comment (sodic) [inline examples/README.md:7]:
```suggestion
The **tutorials** directory contains [Wasp tutorial](https://wasp-lang.dev/docs/tutorial/create) apps.
```

File: examples/README.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+1. **todo-typescript**
+    - A simple task tracking app implemented in TypeScript.
```
Comment (sodic) [inline examples/README.md:10]:
```suggestion
    - A simple task-tracking app implemented in TypeScript.
```

File: examples/README.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+2. **waspello**
+    - A simple Trello clone.
+    - Demonstrates: **[optimistic updates](https://wasp-lang.dev/docs/data-model/operations/actions#the-useaction-hook-and-optimistic-updates)**, [auth](https://wasp-lang.dev/docs/auth/overview), [rpc](https://wasp-lang.dev/docs/data-model/operations/overview)
+
+3. **hackaton-submissions**
+    - An app for organizing your own hackaton.
+    - Demonstrates: [tailwind](https://wasp-lang.dev/docs/project/css-frameworks#tailwind), [rpc](https://wasp-lang.dev/docs/data-model/operations/overview)
+
+4. **waspleau**
+    - A simple clone of Tableau; it regularly pulls in external data and shows it via nice dashboard.
```
Comment (sodic) [inline examples/README.md:22]:
```suggestion
    - A simple clone of Tableau. The app regularly pulls in external data and shows it on a nice dashboard.
```

File: examples/README.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    - Demonstrates: **[jobs](https://wasp-lang.dev/docs/advanced/jobs)**, analytics
+
+5. **thoughts**
+    - A simple note-taking app organized around the concept of hashtags.
+    - Demonstrates: [db seeding](https://wasp-lang.dev/docs/data-model/backends#seeding-the-database), [auth](https://wasp-lang.dev/docs/auth/overview), [rpc](https://wasp-lang.dev/docs/data-model/operations/overview)
+
+6. **realworld**
+    - Medium clone (https://github.com/gothinkster/realworld - mother of all demo apps).
```
Comment (sodic) [inline examples/README.md:30]:
```suggestion
    - A Medium clone (https://github.com/gothinkster/realworld - the mother of all demo apps).
```

File: web/src/components/ExampleWaspApps.js [hunk 8]
```diff
@@ -5,35 +5,33 @@ import { GitHub, ArrowUpRight, Monitor } from 'react-feather'
...
+    repoName: 'cover-letter-gpt',
+    repoUrl: 'https://github.com/vincanger/coverlettergpt',
+    demoUrl: 'https://coverlettergpt.xyz/',
   },
   {
-    title: 'Waspleau 📊',
+    title: 'Realtime voting via WebSockets 🔌',
     description:
-      'A simple data dashboard that makes use of Wasp async jobs feature.',
+      'A realtime, websockets powered voting app built with Wasp in TypeScript.',
```
Comment (sodic) [inline web/src/components/ExampleWaspApps.js:29]:
```suggestion
      'A realtime, websockets-powered voting app built with Wasp and TypeScript.',
```

File: web/src/components/ExampleWaspApps.js [hunk 9]
```diff
@@ -5,35 +5,33 @@ import { GitHub, ArrowUpRight, Monitor } from 'react-feather'
...
   {
-    title: 'Real World App 🐑',
-    description: 'A Medium clone made with Wasp and Material UI.',
-    authorName: 'wasp',
-    authorImg: 'https://avatars.githubusercontent.com/u/55102317',
-    repoName: 'real-world-app',
-    repoUrl: 'https://github.com/wasp-lang/wasp/tree/main/examples/realworld',
-    demoUrl: 'https://wasp-rwa.netlify.app/',
+    title: 'CoverLetterGPT 🤖',
+    description: 'Generate cover letters based on your CV and job description. Powered by ChatGPT.',
```
Comment (sodic) [inline web/src/components/ExampleWaspApps.js:19]:
```suggestion
    description: 'Generate cover letters based on your CV and the job description. Powered by ChatGPT.',
```

**Review by sodic — CHANGES_REQUESTED** [review]
I have no problem with the PR's goal, but that's mostly for @vincanger to comment on (the only reason I'm involved with the PR is merge conflict resolution).  As far as I'm concerned, short names make a lot of sense.

Aside from that, I've corrected a couple of errors and given some suggestions on improving the text.

Before submitting the PR, please go through the style guide (now that we finally have one :)). I think the style guide (and Grammarly, which the style guide prescribes) address all the mistakes and suggestions my comments mention. I know that this PR does not strictly fall under the "docs" category, but it's still something our users read. Therefore, the same rules should probably apply.

**Comment by Martinsos** [comment]
> I have no problem with the PR's goal, but that's mostly for @vincanger to comment on (the only reason I'm involved with the PR is merge conflict resolution). As far as I'm concerned, short names make a lot of sense.
> 
> Aside from that, I've corrected a couple of errors and given some suggestions on improving the text.
> 
> Before submitting the PR, please go through the style guide (now that we finally have one :)). I think the style guide (and Grammarly, which the style guide prescribes) address all the mistakes and suggestions my comments mention. I know that this PR does not strictly fall under the "docs" category, but it's still something our users read. Therefore, the same rules should probably apply.

Thanks for the review!

I didn't get what you mean by "the only reason I'm involved with the PR is merge conflict resolution", from what I saw what you did is fixed/improved grammar/wording, did I miss something?

Guilty here for not using Grammarly, I have to set it up in my editor. As for style guide - I want through all of it when I reviewed it, so I am aware of the concepts in it (although I am sure some I forgot), but it is not clear to me in this PR what I did wrong in regards to those -> it might help if you specifically point out those mistakes that you think went against docs style guide and are not just Grammarly fixes, so I can learn something from this.

**Review by sodic — APPROVED** [review]
All good on my side.

**Comment by vincanger** [comment]
these names sound good and reasonable. My main aim was to make it easier for outsiders to understand what the Apps do, and the README does that.

**Comment by Martinsos** [comment]
> these names sound good and reasonable. My main aim was to make it easier for outsiders to understand what the Apps do, and the README does that.

Awesome, I am glad to hear that! I will then resolve the conflicts and comments and merge.

**Pull request merged**